TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -888,6 +894,32 @@ private async Task<IReadOnlyDictionary<Guid, AuditDetailCollection>> GetAuditRec
             .ToDictionary(t => t.Id, t => t.AuditDetailCollection));
     }
 
+    private async Task<IReadOnlyDictionary<Guid, AuditDetailCollection>> GetAuditRecordsFromAuditRepositoryAsync(
```

Comment:
Don't need chunking here. Something like this will do it:

```cs
private async Task<IReadOnlyDictionary<Guid, AuditDetailCollection>> GetAuditRecordsFromAuditRepositoryAsync(
        string entityLogicalName,
        IEnumerable<Guid> ids,
        CancellationToken cancellationToken)
    {
        return await ids.ToAsyncEnumerable()
            .SelectAwait(async id =>
                (Id: id, Audit: await auditRepository.GetAuditDetailAsync(entityLogicalName, id) ?? throw new Exception($"No audit detail for '{id}'.")))
            .ToDictionaryAsync(k => k.Id, k => k.Audit);
    }
```